raster-test/src/test/scala/geotrellis/raster/rasterize/extent/ExtentRasterizerSpec.scala

```diff
@@ -0,0 +1,106 @@
+ * limitations under the License.
+ */
+
+package geotrellis.raster.rasterize.extent
+
+import geotrellis.raster._
+import geotrellis.vector._
+import geotrellis.raster.rasterize.polygon.PolygonRasterizer
+import geotrellis.testkit._
+
+import math.{max,min,round}
+
+import org.scalatest.FunSuite
+
+class ExtentRasterizerSpec extends FunSuite
+    with TestEngine
+    with TileBuilders {
+
+  test("Rasterization of Covering Extent") {
+    val e = Extent(0.0, 0.0, 10.0, 10.0)
+    val re = RasterExtent(Extent(0.0, 0.0, 10.0, 10.0), 1.0, 1.0, 10, 10)
+    var sum = 0
+    ExtentRasterizer.foreachCellByExtent(e, re) { (x : Int, y : Int) => sum = sum + 1 }
+    assert( sum == 100)
+  }
+
+  test("Rasterization of Partially-Covering Extent w/ includeExterior") {
+    val e = Extent(0.51, 0.51, 9.49, 9.49)
+    val re = RasterExtent(Extent(0.0, 0.0, 10.0, 10.0), 1.0, 1.0, 10, 10)
+    var sum = 0
+    ExtentRasterizer.foreachCellByExtent(e, re, includeExterior = true) { (x : Int, y : Int) => sum = sum + 1 }
+    assert( sum == 100)
+  }
+
+  test("Rasterization of Partially-Covering Extent w/o includeExterior") {
+    val e = Extent(0.51, 0.51, 9.49, 9.49)
+    val re = RasterExtent(Extent(0.0, 0.0, 10.0, 10.0), 1.0, 1.0, 10, 10)
+    var sum = 0
+    ExtentRasterizer.foreachCellByExtent(e, re) { (x : Int, y : Int) => sum = sum + 1 }
+    assert( sum == 64)
+  }
+
+  test("Rasterization of Non-Square Pixels w/ includeExterior") {
+    val e = Extent(1.01, 1.01, 8.99, 8.89)
+    val re = RasterExtent(Extent(0.0, 0.0, 10.0, 10.0), 2.0, 2.0, 10, 10)
+    var sum = 0
+    ExtentRasterizer.foreachCellByExtent(e, re, includeExterior = true) { (x : Int, y : Int) => sum = sum + 1 }
+    assert( sum == 25)
+  }
+
+  test("Rasterization of Non-Square Pixels w/o includeExterior") {
+    val e = Extent(1.01, 1.01, 8.99, 8.89)
+    val re = RasterExtent(Extent(0.0, 0.0, 10.0, 10.0), 2.0, 2.0, 10, 10)
+    var sum = 0
+    ExtentRasterizer.foreachCellByExtent(e, re) { (x : Int, y : Int) => sum = sum + 1 }
+    assert( sum == 9)
+  }
+}
+
+class ExtentPolygonXCheckSpec extends FunSuite
+    with TestEngine
+    with TileBuilders {
+
+  test("Cross-Check with Polygon Rasterizer 1") {
+    val e = Extent(0.51, 0.51, 9.49, 9.49)
+    val re = RasterExtent(Extent(0.0, 0.0, 10.0, 10.0), 1.0, 1.0, 10, 10)
+    var extentSum = 0
+    var polySum = 0
+
+    ExtentRasterizer.foreachCellByExtent(e, re, includeExterior = false) { (x : Int, y : Int) => extentSum = extentSum + 1 }
+    PolygonRasterizer.foreachCellByPolygon(e, re, includeExterior = false) { (x : Int, y : Int) => polySum = polySum + 1 }
+    assert( extentSum == polySum )
+
+    ExtentRasterizer.foreachCellByExtent(e, re, includeExterior = true) { (x : Int, y : Int) => extentSum = extentSum + 1 }
+    PolygonRasterizer.foreachCellByPolygon(e, re, includeExterior = true) { (x : Int, y : Int) => polySum = polySum + 1 }
+    assert( extentSum == polySum )
+  }
+
+  test("Cross-Check with Polygon Rasterizer 2") {
```

Comment:
Is there a more descriptive name you could give these two (like what are they testing differently?)